### DISCUSSION TITLE
Generate a formatting configuration

### DISCUSSION THREAD

**Discussion by delanym** [revelc/formatter-maven-plugin#718] [Q&A] [answered]
Disclaimer: I don't use Eclipse. Is there any way to generate a [configuration file](https://code.revelc.net/formatter-maven-plugin/examples.html#custom-configuration-file) without installing Eclipse and exporting it?

I want to generate a new default configuration file for each [new version](https://code.revelc.net/formatter-maven-plugin/eclipse-versions.html) of Eclipse so I can see what changed/how our coding convention should be adapted.

Exporting Java formatting settings from a new installation of Eclipse results in an empty file:
``` xml
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<profiles version="22"/>
```

**Comment by hazendaz** [comment, marked as answer]
The current default file sorted alphabetically is in the current project.  https://github.com/revelc/formatter-maven-plugin/blob/main/src/main/resources/formatter-maven-plugin/eclipse/java.xml

It's generally updated as needed.  Formatting doesn't always change between releases.

The file exported from eclipse is not sorted and its gui based for setup and very extensive.

That said you either get it here from the jar or have to install eclipse and export it.

Sent from my Verizon, Samsung Galaxy smartphone
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
From: Delany ***@***.***>
Sent: Wednesday, March 22, 2023 4:53:02 PM
To: revelc/formatter-maven-plugin ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [revelc/formatter-maven-plugin] Generate a formatting configuration (Discussion #718)


Disclaimer: I don't use Eclipse. Is there any way to generate a configuration file<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fcode.revelc.net%2Fformatter-maven-plugin%2Fexamples.html%23custom-configuration-file&data=05%7C01%7C%7C06ad3cc8b2c24106276d08db2b176e6c%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C638151151866813813%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=IopQHjW%2B1S7z%2FGETjseZEe8CTUbIIjLFqfLdhx3L3%2FU%3D&reserved=0> without installing Eclipse and exporting it?

I want to generate a new default configuration file for each new version<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fcode.revelc.net%2Fformatter-maven-plugin%2Feclipse-versions.html&data=05%7C01%7C%7C06ad3cc8b2c24106276d08db2b176e6c%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C638151151866813813%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=M9B2rFWz3%2B1Gjp6m2fLq8Fk68Colyh5rdYsc51F2PVs%3D&reserved=0> of Eclipse so I can see what changed/how our coding convention should be adapted.

**Reply by delanym** [reply]
This is very helpful thank you.

**Comment by hazendaz** [comment]
Also sort is separate.  Eclipse export is dynamic in nature so it changes.  Generally use excel to sort the results.

Sent from my Verizon, Samsung Galaxy smartphone
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
From: Delany ***@***.***>
Sent: Wednesday, March 22, 2023 4:53:02 PM
To: revelc/formatter-maven-plugin ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [revelc/formatter-maven-plugin] Generate a formatting configuration (Discussion #718)


Disclaimer: I don't use Eclipse. Is there any way to generate a configuration file<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fcode.revelc.net%2Fformatter-maven-plugin%2Fexamples.html%23custom-configuration-file&data=05%7C01%7C%7C06ad3cc8b2c24106276d08db2b176e6c%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C638151151866813813%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=IopQHjW%2B1S7z%2FGETjseZEe8CTUbIIjLFqfLdhx3L3%2FU%3D&reserved=0> without installing Eclipse and exporting it?

I want to generate a new default configuration file for each new version<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fcode.revelc.net%2Fformatter-maven-plugin%2Feclipse-versions.html&data=05%7C01%7C%7C06ad3cc8b2c24106276d08db2b176e6c%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C638151151866813813%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=M9B2rFWz3%2B1Gjp6m2fLq8Fk68Colyh5rdYsc51F2PVs%3D&reserved=0> of Eclipse so I can see what changed/how our coding convention should be adapted.
